### ISSUE TITLE
How can I use websockets with C#?

### ISSUE THREAD

**Issue by timeshift92** [dotnet-websharper/aspnetcore#14]
can you example please

**Comment by granicz** [comment]
There is an [F# test project for web sockets](https://github.com/dotnet-websharper/aspnetcore/tree/master/WebSharper.AspNetCore.WebSocket) and a couple other examples on the web - but indeed, we don't seem to have anything for C#. @Jand42 are you aware of any? If not, we should add a C# test project for websockets in this repo.

**Comment by Jand42** [comment]
@granicz Sadly there is a good reason why we don't have a C# test yet - the API we have is _very_ F# oriented. To use from C# as is you would need a bunch of FSharpFunc/FSharpAsync conversions, basically try to replicate exactly what the F# code does. So the better first step here is to create a C#-oriented version of the API with System.Funcs/Tasks.

**Comment by timeshift92** [comment]
yes it is, I spent almost two days here to convert the example to C #, I even used chat gpt, but to no avail

**Comment by timeshift92** [comment]
And can you run this example?
https://github.com/websharper-samples/WebsocketChart

After running and generating number for chart, click f5 (reload page) websocket get error

**Comment by granicz** [comment]
Yeah, looks like that sample doesn't quite manage state changes properly. Any takers to update it to correct that?

**Comment by granicz** [comment]
cc/ @diegopego @Jooseppi12 - This would be a semi-good example for the new documentation too.
